nrfx_config_nrf9160.h

```diff
@@ -36,243 +36,83 @@
-#endif
-
-#ifdef DT_NORDIC_NRF_RTC_RTC_0_BASE_ADDRESS
-#define NRF_RTC0 \
-        ((NRF_RTC_Type *)DT_NORDIC_NRF_RTC_RTC_0_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_RTC_RTC_1_BASE_ADDRESS
-#define NRF_RTC1 \
-        ((NRF_RTC_Type *)DT_NORDIC_NRF_RTC_RTC_1_BASE_ADDRESS)
-#endif
+#if defined(CONFIG_ARM_NONSECURE_FIRMWARE)
+#define NRF_PERIPH(P)   P##_NS
+#else
+#define NRF_PERIPH(P)   P##_S
+#endif
+
+#define NRF_CLOCK       NRF_PERIPH(NRF_CLOCK)
+#define NRF_CRYPTOCELL  NRF_PERIPH(NRF_CRYPTOCELL)
+#define NRF_DPPIC       NRF_PERIPH(NRF_DPPIC)
+#define NRF_EGU0        NRF_PERIPH(NRF_EGU0)
+#define NRF_EGU1        NRF_PERIPH(NRF_EGU1)
+#define NRF_EGU2        NRF_PERIPH(NRF_EGU2)
+#define NRF_EGU3        NRF_PERIPH(NRF_EGU3)
+#define NRF_EGU4        NRF_PERIPH(NRF_EGU4)
+#define NRF_EGU5        NRF_PERIPH(NRF_EGU5)
+#define NRF_FICR        NRF_PERIPH(NRF_FICR)
+#define NRF_FPU         NRF_PERIPH(NRF_FPU)
+#define NRF_IPC         NRF_PERIPH(NRF_IPC)
+#define NRF_I2S         NRF_PERIPH(NRF_I2S)
+#define NRF_KMU         NRF_PERIPH(NRF_KMU)
+#define NRF_NVMC        NRF_PERIPH(NRF_NVMC)
+#define NRF_P0          NRF_PERIPH(NRF_P0)
+#define NRF_PDM         NRF_PERIPH(NRF_PDM)
+#define NRF_POWER       NRF_PERIPH(NRF_POWER)
+#define NRF_PWM0        NRF_PERIPH(NRF_PWM0)
+#define NRF_PWM1        NRF_PERIPH(NRF_PWM1)
+#define NRF_PWM2        NRF_PERIPH(NRF_PWM2)
+#define NRF_PWM3        NRF_PERIPH(NRF_PWM3)
+#define NRF_REGULATORS  NRF_PERIPH(NRF_REGULATORS)
+#define NRF_RTC0        NRF_PERIPH(NRF_RTC0)
+#define NRF_RTC1        NRF_PERIPH(NRF_RTC1)
+#define NRF_SAADC       NRF_PERIPH(NRF_SAADC)
+#define NRF_SPIM0       NRF_PERIPH(NRF_SPIM0)
+#define NRF_SPIM1       NRF_PERIPH(NRF_SPIM1)
+#define NRF_SPIM2       NRF_PERIPH(NRF_SPIM2)
+#define NRF_SPIM3       NRF_PERIPH(NRF_SPIM3)
+#define NRF_SPIS0       NRF_PERIPH(NRF_SPIS0)
+#define NRF_SPIS1       NRF_PERIPH(NRF_SPIS1)
+#define NRF_SPIS2       NRF_PERIPH(NRF_SPIS2)
+#define NRF_SPIS3       NRF_PERIPH(NRF_SPIS3)
+#define NRF_SPU         NRF_PERIPH(NRF_SPU)
+#define NRF_TIMER0      NRF_PERIPH(NRF_TIMER0)
+#define NRF_TIMER1      NRF_PERIPH(NRF_TIMER1)
+#define NRF_TIMER2      NRF_PERIPH(NRF_TIMER2)
+#define NRF_TWIM0       NRF_PERIPH(NRF_TWIM0)
+#define NRF_TWIM1       NRF_PERIPH(NRF_TWIM1)
+#define NRF_TWIM2       NRF_PERIPH(NRF_TWIM2)
+#define NRF_TWIM3       NRF_PERIPH(NRF_TWIM3)
+#define NRF_TWIS0       NRF_PERIPH(NRF_TWIS0)
+#define NRF_TWIS1       NRF_PERIPH(NRF_TWIS1)
+#define NRF_TWIS2       NRF_PERIPH(NRF_TWIS2)
+#define NRF_TWIS3       NRF_PERIPH(NRF_TWIS3)
+#define NRF_UARTE0      NRF_PERIPH(NRF_UARTE0)
+#define NRF_UARTE1      NRF_PERIPH(NRF_UARTE1)
+#define NRF_UARTE2      NRF_PERIPH(NRF_UARTE2)
+#define NRF_UARTE3      NRF_PERIPH(NRF_UARTE3)
+#define NRF_UICR        NRF_PERIPH(NRF_UICR)
+#define NRF_VMC         NRF_PERIPH(NRF_VMC)
+#define NRF_WDT         NRF_PERIPH(NRF_WDT)
+
+
+#if defined(CONFIG_ARM_NONSECURE_FIRMWARE)
```

Comment:
This part about GPIOTE would not be needed, if we used the DTS-way to define these. Because we have already defined different GPIOTEs in DTS for the two S and NS board versions

Reply:
I agree, but the overall point still stands. This is the "nrfx way"of configuring nrfx, and I think it's more consistent with other nrfx uses.

Reply:
> This is to configure nrfx, not Zephyr. nrfx comes with a hardcoded list of peripherals and their addresses, and I think it makes sense to use those. If we ever have a mismatch between DT and nrfx then it will clearly show as the peripheral not working. I am in favor thus of using nrfx macros to configure nrfx and DT to configure the Zephyr shim layers.

But Zephyr drivers, now depend on this, to get the peripheral base addresses. So it seems to me that nRF drivers in Zephyr get #define macros from both nRFx and DTS in the source code. Can we clean this up?